talk/morelanguage/constexpr.tex

```diff
@@ -150,7 +150,7 @@
   \begin{block}{Motivation}
     \begin{itemize}
     \item Like a \mintinline{cpp}{constexpr} variable, a \mintinline{cpp}{constinit} variable guarantees compile-time initialization, but can be modified afterwards
-    \item Only allowed for \mintinline{cpp}{static}/global and \mintinline{cpp}{thread_local} variables
+    \item Makes variable \mintinline{cpp}{static}/global or \mintinline{cpp}{thread_local}
```

Comment:
This is not correct. `constinit` does not make a non-static variable `static`. E.g. This code:
```c++
int f() {
    constinit int a = 3; // error
}
struct S {
    constinit int a = 4; // error
};
```
Has two compile errors. So the original sentence was more correct. The `constinit` keyword is only allowed on a variable which is already `static`/global or `thread_local`.

Reply:
hum... I've reread cppreference about that in the afternoon, and the very first sentence for constinit is : "The constinit specifier declares a variable with static or thread storage duration". See https://en.cppreference.com/w/cpp/language/constinit

And the example given does not use any static or anything.

Reply:
> And the example given does not use any static or anything.

The example on the slide is fine because the variables `a`, `b` and `c` are in global scope and thus have static storage duration.

Reply:
ok, let me revert that too then